app/src/main/java/app/insti/fragment/DetailedComplaintFragment.java

```diff
@@ -0,0 +1,307 @@
+        commentListAdapter.notifyItemRangeChanged(0, commentListAdapter.getItemCount());
+        recyclerViewComments.post(new Runnable() {
+            @Override
+            public void run() {
+                MainActivity.hideKeyboard(getActivity());
+            }
+        });
+    }
+
+    public void setDetailedComplaint(Venter.Complaint detailedComplaint) {
+        this.detailedComplaint = detailedComplaint;
+        populateViews();
+    }
+
+    private void populateViews() {
+        try {
+            textViewUserName.setText(detailedComplaint.getComplaintCreatedBy().getUserName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            String time = DateTimeUtil.getDate(detailedComplaint.getComplaintReportDate().toString());
+            Log.i(TAG, " time: " + time);
+            Log.i(TAG, "inside try");
+            textViewReportDate.setText(time);
+
+        } catch (Exception e) {
+            Log.i(TAG, "Inside catch");
+            e.printStackTrace();
+        }
+        try {
+            textViewLocation.setText(detailedComplaint.getLocationDescription());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            textViewDescription.setText(detailedComplaint.getDescription());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            textViewStatus.setText(detailedComplaint.getStatus().toUpperCase());
+            if (detailedComplaint.getStatus().equalsIgnoreCase("Reported")) {
+                textViewStatus.setBackgroundColor(Color.parseColor("#FF0000"));
+                textViewStatus.setTextColor(getResources().getColor(R.color.primaryTextColor));
+            } else if (detailedComplaint.getStatus().equalsIgnoreCase("In Progress")) {
+                textViewStatus.setBackgroundColor(getResources().getColor(R.color.colorSecondary));
+                textViewStatus.setTextColor(getResources().getColor(R.color.secondaryTextColor));
+            } else if (detailedComplaint.getStatus().equalsIgnoreCase("Resolved")) {
+                textViewStatus.setBackgroundColor(Color.parseColor("#00FF00"));
+                textViewStatus.setTextColor(getResources().getColor(R.color.secondaryTextColor));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            textViewCommentLabel.setText("COMMENTS");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            textViewVoteUpLabel.setText("UP VOTES");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            addVotesToView(detailedComplaint);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
+            addCommentsToView(detailedComplaint);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Again, so many try catch blocks are ugly

Reply:
ok. will put in single try catch